flink-connector-pulsar/src/main/java/org/apache/flink/connector/pulsar/source/reader/deserializer/PulsarDeserializationSchema.java

```diff
@@ -70,80 +66,4 @@ default void open(InitializationContext context, SourceConfiguration configurati
      * @param out The collector to put the resulting messages.
      */
     void deserialize(Message<byte[]> message, Collector<T> out) throws Exception;
-
-    /**
-     * Create a PulsarDeserializationSchema by using the flink's {@link DeserializationSchema}. It
-     * would consume the pulsar message as the byte array and decode the message by using flink's
-     * logic.
-     *
-     * @deprecated Use {@link PulsarSourceBuilder#setDeserializationSchema(DeserializationSchema)}
-     *     instead.
-     */
-    @Deprecated
-    static <T> PulsarDeserializationSchema<T> flinkSchema(
```

Comment:
... while these methods are not buggy. While it's possible to remove in the next major version, I wonder why we do this change and it'd better we prevent further changes like such rename things.

Reply:
Let me have an explain on this refactoring. I just deprecated and removed these methods in `PulsarDeserializationSchema` and `PulsarSerializationSchema` for following the API annotation rules. Since the `PulsarSerializationSchema` and `PulsarDeserializationSchema` are annotated with `@PublicEvolving`. All the methods in this class should be public viewed and can be used without any worry. But the methods like `flinkSchema` returned an internal implementation of `Pulsar(De)serializationSchema` this is not acceptable and may be hard to change in the future.

So I just move these to the source/sink builder method for uniform design style and hide these internal implementations (they are annotated with `@Internal`). This change also made it easy to implement add the methods in `flink-python` module.